packages/core/src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -138,15 +146,56 @@ const EditableTypography: VibeComponent<EditableTypographyProps, HTMLElement> =
       }
     }
 
-    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    function handleChange(event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
       setInputValue(event.target.value);
+
+      if (multiline) {
+        resizeTextarea();
+      }
     }
 
     const toggleKeyboardEditMode = useKeyboardButtonPressedFunc(toggleEditMode);
 
     function focus() {
       if (inputRef.current) {
         inputRef.current?.focus();
+
+        if (multiline) {
+          resizeTextarea();
+        }
+      }
+    }
+
+    /* Dynamically resizes the textarea to fit its content */
+    function resizeTextarea() {
+      console.log("*************************resize called, why?", multiline, inputRef.current);
+      if (inputRef.current) {
+        // Temporarily set the height to "auto" to accurately measure the scroll height of the content inside the textarea.
+        setInputHeight("auto");
```

Comment:
This might cause layout thrashing

Reply:
i am requesting an animation frame to decrease the chances of this to happen